### ISSUE TITLE
update thingsboard failed Version 3.6.2PE to 3.6.3PE (Google Cloud)

### ISSUE THREAD

**Issue by xtaveira** [thingsboard/thingsboard#10962]
When I try to start the thingsboard service after the update, I got this fail

```
thingsboard.service - thingsboard
     Loaded: loaded (/lib/systemd/system/thingsboard.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Mon 2024-06-10 17:53:58 UTC; 956ms ago
    Process: 70618 ExecStart=/usr/share/thingsboard/bin/thingsboard.jar (code=exited, status=1/FAILURE)
   Main PID: 70618 (code=exited, status=1/FAILURE)

Jun 10 17:53:31 thingsboard-pe-1-vm thingsboard.jar[70633]:  ===================================================
Jun 10 17:53:31 thingsboard-pe-1-vm thingsboard.jar[70633]:  :: ThingsBoard ::       (v3.6.3)
Jun 10 17:53:31 thingsboard-pe-1-vm thingsboard.jar[70633]:  ===================================================
Jun 10 17:53:50 thingsboard-pe-1-vm thingsboard.jar[70633]: WARNING: An illegal reflective access operation has occurred
Jun 10 17:53:50 thingsboard-pe-1-vm thingsboard.jar[70633]: WARNING: Illegal reflective access by org.nustaq.serialization.FSTClazzInfo (jar:file:/usr/share/thingsboard/bin/thingsboard.jar>
Jun 10 17:53:50 thingsboard-pe-1-vm thingsboard.jar[70633]: WARNING: Please consider reporting this to the maintainers of org.nustaq.serialization.FSTClazzInfo
Jun 10 17:53:50 thingsboard-pe-1-vm thingsboard.jar[70633]: WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
Jun 10 17:53:50 thingsboard-pe-1-vm thingsboard.jar[70633]: WARNING: All illegal access operations will be denied in a future release
Jun 10 17:53:58 thingsboard-pe-1-vm systemd[1]: thingsboard.service: Main process exited, code=exited, status=1/FAILURE
Jun 10 17:53:58 thingsboard-pe-1-vm systemd[1]: thingsboard.service: Failed with result 'exit-code'.
```

and in the logs I got this error

```
Caused by: org.postgresql.util.PSQLException: ERROR: column componentd0_.has_queue_name does not exist
  Position: 301
```

```
at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:103)
at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
```

I tried to upgrade my thingsboard version following the documentation(the database backup was made correctly), also tried to deactivate my instance deleting the file "/usr/share/thingsboard/bin/instance-license.data" and I'm checking the license ID after the update.I don't want to migrate my database, I want to stay with the currently one.

**Comment by pon0marev** [comment]
@xtaveira Open the file `/usr/share/thingsboard/data/upgrade/3.6.2/schema_update.sql`
and run queries to update the database schema manually. Then, restart the service.

**Comment by xtaveira** [comment]
> @xtaveira Open the file `/usr/share/thingsboard/data/upgrade/3.6.2/schema_update.sql` and run queries to update the database schema manually. Then, restart the service.

![image](https://github.com/thingsboard/thingsboard/assets/96428113/a1d4eb4f-9e3d-44ad-9723-a90a51afc955)

There is no 3.6.2 directory on my upgrade folder, but it is on the 3.6.2PE by default.

![image](https://github.com/thingsboard/thingsboard/assets/96428113/8726b83a-43ad-4750-8351-375628349d60)

**Comment by pon0marev** [comment]
Re-download and reinstall version 3.6.3. You should definitely have the 3.6.2 folder after unpacking version 3.6.3.
```bash
wget https://dist.thingsboard.io/thingsboard-3.6.3pe.deb
sudo dpkg -i thingsboard-3.6.3pe.deb
```

If still, for some reason, you don't have this folder available, here is the script for updating the database schema. Please be sure to make a backup copy of the database before running the next queries.
```sql
DROP INDEX IF EXISTS idx_rule_node_type;
DROP INDEX IF EXISTS idx_rule_node_type_configuration_version;
CREATE INDEX IF NOT EXISTS idx_rule_node_type_id_configuration_version ON rule_node(type, id, configuration_version);

-- RULE NODE INDEXES UPDATE END

-- RULE NODE QUEUE UPDATE START

ALTER TABLE rule_node ADD COLUMN IF NOT EXISTS queue_name varchar(255);
ALTER TABLE component_descriptor ADD COLUMN IF NOT EXISTS has_queue_name boolean DEFAULT false;

-- RULE NODE QUEUE UPDATE END

-- SCHEDULER EVENT UPDATE START

ALTER TABLE scheduler_event ADD COLUMN IF NOT EXISTS enabled boolean default true;

-- SCHEDULER EVENT UPDATE END


DO
$$
    BEGIN
        IF NOT EXISTS(SELECT 1 FROM information_schema.columns WHERE table_name = 'user_settings' AND column_name = 'settings' AND data_type = 'jsonb') THEN
            ALTER TABLE user_settings RENAME COLUMN settings to old_settings;
            ALTER TABLE user_settings ADD COLUMN settings jsonb;
            UPDATE user_settings SET settings = old_settings::jsonb WHERE old_settings IS NOT NULL;
            ALTER TABLE user_settings DROP COLUMN old_settings;
        END IF;
    END;
$$;
```

**Comment by xtaveira** [comment]
I followed these instructions and it solved my problem, I also updated to the 3.6.4PE version. thanks @pon0marev! :D
![image](https://github.com/thingsboard/thingsboard/assets/96428113/b20f44af-2393-41de-8a14-bf0fb023c5ee)
